gateway/analytics.go

```diff
@@ -53,6 +53,10 @@ type RedisAnalyticsHandler struct {
 	Gw                          *Gateway `json:"-"`
 	mu                          sync.Mutex
 	analyticsSerializer         serializer.AnalyticsSerializer
+
+	// testing purposes
+	mockEnabled   bool
```

Comment:
we can think of having a `MockRedisAnalyticsHandler` to expect the mock response, may be part of a later refactor

Reply:
@jeffy-mathew I have tried it actually but `RedisAnalyticsHandler` has lots of field usages, the interface will have a lot of getter methods and the codebase will be changed with those. I know this way is not the preferred way but it was the simplest. That's why I preferred.